### ISSUE TITLE
F-Droid updates stuck

### ISSUE THREAD

**Issue by IzzySoft** [rosuH/EasyWatermark#31]
Updates of your app at the [F-Droid](https://f-droid.org/) repository are currently stuck. Reason is the updater no longer can identify new versions, as you're using variables for `versionCode` and `versionName` [in your `build.gradle`](https://github.com/rosuH/EasyWatermark/blob/1.1.2/app/build.gradle.kts#L21) – but the updater needs the "real values" there (it just parses the `build.gradle`, it does not execute `gradlew`).

If you want F-Droid to pick up again,

```gradle
        versionCode = Apps.versionCode
        versionName = Apps.versionName
```

must be changed to reflect the real values, currently this would be

```gradle
        versionCode = 10102
        versionName = "1.0.3"
```

**Comment by IzzySoft** [comment]
Looks good, thanks! But will only take effect with the next tag, as that's where the updater looks at. You could of course delete the `1.1.2` tag & recreate it at HEAD.

**Comment by rosuH** [comment]
The `1.1.2` has been re-published. Thank you for your guidance.

**Comment by linsui** [comment]
This problem happens again. Could you please use static version code? I have update it manually. Thanks!

**Comment by rosuH** [comment]
> This problem happens again. Could you please use static version code? I have update it manually. Thanks!

I will check it asap 👌

**Comment by rosuH** [comment]
@linsui Hello linsui, I noticed you have modified the `UpdateCheckMode` to HTTP in this [fdroiddata](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/21455) merge request. After reviewing the docs, it appears updates are now checked by inspecting the release apk.

Running `fdroid checkupdates me.rosuh.easywatermark`, I confirmed that updates are detected.

```
╰─$ fdroid checkupdates me.rosuh.easywatermark
2025-04-12 09:49:43,194 INFO: Using /Users/rosu/Library/Android/sdk/build-tools/35.0.0/apksigner
2025-04-12 09:49:44,882 INFO: Processing me.rosuh.easywatermark
2025-04-12 09:49:49,465 INFO: ...updating to version 2.9.5 (20905)
```

Do I still need to set `versionName` and `versionCode` to [static value](https://gitlab.com/fdroid/wiki/-/wikis/FAQ#why-do-we-always-ask-for-static-values-on-versioncodeversionname-in-buildgradle)? Or is maintaining the release APK naming convention sufficient?

**Comment by linsui** [comment]
Yep, but I'd rather avoid using GitHub api so it would be better if you can set versionCode to static value.

**Comment by linsui** [comment]
Thanks!
